pkg/api-server/customization/api_manager.go

```diff
@@ -0,0 +1,41 @@
+package customization
+
+import (
+	"github.com/emicklei/go-restful"
+)
+
+type APIInstaller interface {
+	Install(container *restful.Container)
+}
+
+type APIManager interface {
+	APIInstaller
+	Add(ws *restful.WebService)
+	AddRouteFunction(path, subpath string, handler restful.RouteFunction)
+}
+
+type APIList struct {
+	list []*restful.WebService
+}
+
+func NewAPIList() *APIList {
+	return &APIList{
+		list: []*restful.WebService{},
+	}
+}
+
+func (c *APIList) Add(ws *restful.WebService) {
+	c.list = append(c.list, ws)
+}
+
+func (c *APIList) AddRouteFunction(path, subpath string, handler restful.RouteFunction) {
+	ws := new(restful.WebService).Path(path)
+	ws.Route(ws.GET(subpath).To(handler))
+	c.Add(ws)
+}
```

Comment:
What will happen if you need something like Auth on this endpoint or filter of logging requests or add an extra method to this webservice?

I think I'd drop this method and do these 3 lines on the APIManager client that will just use `Add`. That's not really extensible API, you cannot add filters, change the method, add many methods to ws, add OpenAPI docs etc.

Reply:
This is just a convenience method, some will prefer this for simplicity and some to supply a full-featured WS. I don't see what is wrong with having this one here.